modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/utils/AuthenticatorUtil.java

```diff
@@ -65,86 +75,101 @@ private AuthenticatorUtil() {
+                authenticationContext.setDomainAwareUsername(domainAwareUserName);
+
+                // Validation for the admin user of the domain.
+                UserStoreManager userstoremanager =
+                        CarbonContext.getThreadLocalCarbonContext().getUserRealm().getUserStoreManager();
+                String[] userRoles = userstoremanager.getRoleListOfUser(domainAwareUserName);
+                String adminRoleName = CarbonContext.getThreadLocalCarbonContext().getUserRealm()
+                        .getRealmConfiguration().getAdminRoleName();
                 for (String userRole : userRoles) {
                     if (adminRoleName.equalsIgnoreCase(userRole)) {
                         log.info(username + " is authorized to import and export APIs");
-                        return Response.ok().build();
+                        return Response.ok().entity(authenticationContext).build();
                     }
                 }
 
-                if (authorizationManager.isUserAuthorized(tenantAwareUsername, APIM_ADMIN_PERMISSION,
-                        CarbonConstants.UI_PERMISSION_ACTION)) {
+                // Validation for a user having API-M Admin, API Create and Login permissions.
+                AuthorizationManager authorizationManager = CarbonContext.getThreadLocalCarbonContext().getUserRealm()
+                        .getAuthorizationManager();
+                if ((authorizationManager.isUserAuthorized(domainAwareUserName, APIM_ADMIN_PERMISSION,
+                        CarbonConstants.UI_PERMISSION_ACTION)) && (authorizationManager.isUserAuthorized
+                        (domainAwareUserName, APIM_LOGIN_PERMISSION, CarbonConstants.UI_PERMISSION_ACTION)) &&
+                        (authorizationManager.isUserAuthorized(domainAwareUserName, APIM_API_CREATE_PERMISSION,
+                                CarbonConstants.UI_PERMISSION_ACTION))) {
                     log.info(username + " is authorized to import and export APIs");
-                    return Response.ok().build();
+                    return Response.ok().entity(authenticationContext).build();
                 }
 
-                return Response.status(Response.Status.FORBIDDEN).entity("User is not authorized for the " +
-                        "performed action.").type(MediaType.APPLICATION_JSON).build();
+                return Response.status(Response.Status.FORBIDDEN).entity("User Authorization Failed")
+                        .type(MediaType.APPLICATION_JSON).build();
 
             } else {
                 return Response.status(Response.Status.UNAUTHORIZED).entity("User Authentication Failed")
                         .type(MediaType.APPLICATION_JSON).build();
             }
-
-        } catch (UserStoreException e) {
-            String errorMessage = "Error while accessing user configuration";
+        } catch (Exception e) {
```

Comment:
Any reason to catch root Exception? Is there any methods in the try block throwing Exception?

Reply:
All of the following exceptions needs to be caught. 
APIExportException, RemoteException, MalformedURLException, LoginAuthenticationExceptionException, ExceptionException. That is the reason why I have caught the root exception. Multi catch cannot be used due to the language level dependency (1.6)